app/models/experiment/reminder_experiment.rb

```diff
@@ -0,0 +1,7 @@
+module Experiment
```

Comment:
The stated possibility that we may do other experiments led me to put these in a namespace and include the experiment domain (reminders) in the name to avoid squatting on the name "experiment".

Reply:
> It's very possible we'll never create more experiment types.
Agree, but imo we should try to model `Experiment` as a first class AR backed Model. We'll have to make a tradeoff on:
- keep `Experiment` as generic as possible
- avoid adding too much modelling bloat at this stage

Reply:
Not sure I follow. Are you suggesting changing `Experiment::ReminderExperiment` to just `Experiment`? Is the idea that we could later use single table inheritance if we add more models?

Reply:
I didn't have a specific solution in mind, but I like that approach.

Reply:
> in the name to avoid squatting on the name "experiment".

I don't think any other model should use the name experiment. It should refer to one thing ideally: experiments. And experiments should be separate from what we're doing inside them.

An Experiment should incorporate the follow concerns:
- Have a unique ID. If we have 2 experiments running at the same time, this ID would differentiate them.
- Be immutable: can't perform updates except to activate and inactivate once only
- Treatments (the right term for "buckets"): what they are, and what the traffic split between them is
- Start and end times, if we ever do scheduling
- The consistent hashing algorithm: can be different for different experiments, but perhaps they can be passed in.

We shouldn't [complect](https://www.merriam-webster.com/dictionary/complect) this concept with Reminders, which are how the Treatments are _executed_. This allows experiments to be reusable across others concerns.